aspnetcore/grpc/authn-and-authz.md

```diff
@@ -103,70 +102,76 @@ private static GrpcChannel CreateAuthenticatedChannel(string address)
 
 #### Bearer token with gRPC client factory
 
-gRPC client factory can create clients that send a bearer token using `ChannelCredentials`. When configuring a client, assign the `CallCredentials` the client should use with the `ConfigureChannel` method.
+gRPC client factory can create clients that send a bearer token using `AddCallCredentials`. The delegate passed to `AddCallCredentials` is executed for each gRPC call.
 
 ```csharp
 builder.Services
     .AddGrpcClient<Greeter.GreeterClient>(o =>
     {
         o.Address = new Uri("https://localhost:5001");
     })
-    .ConfigureChannel(o =>
+    .AddCallCredentials((context, metadata) =>
     {
-        var credentials = CallCredentials.FromInterceptor((context, metadata) =>
+        if (!string.IsNullOrEmpty(_token))
         {
-            if (!string.IsNullOrEmpty(_token))
-            {
-                metadata.Add("Authorization", $"Bearer {_token}");
-            }
-            return Task.CompletedTask;
-        });
-
-        o.Credentials = ChannelCredentials.Create(new SslCredentials(), credentials);
+            metadata.Add("Authorization", $"Bearer {_token}");
```

Comment:
It's *waaaay* not related to the updates here, but it's on my mind from some Blazor work that I was performing this morning ...

The underscores for fields, if that's what the `_token` is here, were supposed to go away over time. **_I only mention this in passing._** I placed the cross-links to the discussion and prior work in this area on the .NET 7 issue for cleaning up web templates ...

https://github.com/dotnet/aspnetcore/issues/30162#issuecomment-1108396631